### ISSUE TITLE
Uplift DMN editor

### ISSUE THREAD

**Issue by harmeet-status** [sartography/spiff-arena#737]
Can we build a better DMN editor? Trisotech seems to do it very well. Would love to have a day or two just to research this area and see if there is just another open source project we can leverage, or if we could build our own tool.

One thing to consider here is that DMN involves more than just tables, though we currently only support tables. If we were to build our own tool, we might eventually have to reimplement the non-table components that are already implemented in bpmn-js.

**RESEARCH TASK**

**Comment by harmeet-status** [comment]
https://spiff-arena.readthedocs.io/en/latest/wish_list/wish_list.html#dmn-editor-sucks

https://www.notion.so/Search-DMN-table-using-search-box-131212706e954dc8a3d088f0ba85a58a?pvs=4

**Comment by harmeet-status** [comment]
@danfunk can we use ONLYOFFICE? As we are going to use NextCloud in the organisation which also uses ONLYOFFICE.

**Comment by calexh-sar** [comment]
Here are some Trisotech links with examples if their DMN implementation:

- [Practical DMN: The Basics](https://www.trisotech.com/practical-dmn-the-basics/)
- [DMN 101](https://www.trisotech.com/dmn-101/)
- [What's new in DMN 1.4 and 1.5](https://www.trisotech.com/whats-new-in-dmn-1-4-and-1-5/)
- [DMN: Data Validation Reconsidered](https://www.trisotech.com/dmn-data-validation-reconsidered/)
- [More on-DMN Validation](https://www.trisotech.com/more-on-dmn-validation/)

**Comment by essweine** [comment]
I spent some time looking at this and it is going to be a fair amount of work to make any customizations or to replace it with anything new.  The other thing that became clear to me is that Alex's expectation for the DMN editor have nothing to do with the UI -- they would involve extending our DMN engine.  So we should create different tickets for these, because they are independent of each other.

This is a summary of what I found: https://docs.google.com/document/d/1uh67519nCKJ9CydlUZzL9X_K7TqUBmuqWay8VmsQC9U/edit?usp=sharing
